### PR TITLE
More fishing spots, fix manta and trench

### LINKED ISSUE #3475: Cannot fish on Manta or Trench seafloor
Submitter: studenterhue
**Describe the bug**
I could not use the fishing rod on seafloor tiles surrounding Manta or the seafloor tiles in the Trench. The fishing rod sprite did not change, and there was no "(user) starts fishing" message associated with successful fishing starts. Furthermore, after waiting, I did not catch any fish, so it appears it is not an issue of the sprites not updating/message showing.

**To Reproduce**
0. Be on Manta OR be in Trench
1. Click on an adjacent seafloor tile while holding a fishing rod.

**Expected behavior**
For me to mine 4 fish like on Oshan station z.

**Additional context**
Fishing on Oshan station z-level seafloor works as normal, as well as /turf/space/fluid tiles spawned in. The sprite changes, the "starts fishing" message shows, and I was eventually able to reel in a catch, with the expected reeling message and uniquely drawn fish.

Reported on: Goonstation 2021-01-26 01:54:48

### PR DESCRIPTION
**Pull request by Camryn-Buttes**
[FEATURE] [BUG]
## About the PR 
Adds a fuckload more places to fish using the fishing rod.
Fixes #3475
Adds an ultra-cool testing rod for admins to fish more quickly.

## Why's this needed? 

It's cool.
Fish go swimmy.

## Changelog

```changelog
(u)Camryn Buttes
(+)Added fuckloads more places to fish. Fixed manta and trench fishing.
```

### CODE REVIEW

File: code/modules/fishing/fishing_spots.dm [hunk 1]
```diff
@@ -103,3 +148,108 @@ ABSTRACT_TYPE(/datum/fishing_spot)
...
+	/obj/item/fish/salmon = 20,\
+	/obj/item/fish/herring = 15,\
+	/obj/item/fish/red_herring = 5,\
+	/obj/item/beach_ball = 3,\
+	/obj/item/clothing/shoes/flippers = 3,\
+	/obj/item/clothing/gloves/water_wings = 3,\
+	/obj/item/inner_tube/random = 3,\
+	/obj/item/reagent_containers/food/drinks/curacao = 3,\
+	/obj/item/rubberduck = 2,\
+	/obj/item/organ/chest = 1,\
```
Comment (zjdtmkhzt) [inline code/modules/fishing/fishing_spots.dm:198]: We have an actual treasure chest in the code, but you choose to do this. 
smhing my head v_v

**Comment by virvatuli-gh** [comment]
To link an issue that the PR will fix, put `Fixes #3475` in the message body. This will autoclose the issue if the PR gets merged. The way you have it set up now doesn't actually link to the issue.

File: code/modules/fishing/fishing_spots.dm (hunk 1 above)
Comment (NoahButtes) [reply code/modules/fishing/fishing_spots.dm:198]: I thought it was funnier

**Comment by Adharainspace** [comment]
dnm until i finish my latest fishing patch and push it

**Comment by Camryn-Buttes** [comment]
Any estimate on when the new fishing patch will be pushed?

**Comment by Camryn-Buttes** [comment]
> dnm until i finish my latest fishing patch and push it

It's been about a month, any updates?

**Comment by Adharainspace** [comment]
yes im lazy and slow at working on things and am adding a lot of content onto fishing please be patient i will mark as organic

**Comment by Camryn-Buttes** [comment]
Since Adhara has decided to take an indefinite break from goon, where does that leave this PR?

**Comment by Adharainspace** [comment]
im back baby, this pr is never getting merged because its more work for me and im lazy!!! 
youll have to wait until i figure out why fishing rods BROKE on my branch!!!!

**Comment by Camryn-Buttes** [comment]
Yeah, I'm gonna go ahead and close this and remake some of the changes in more atomized PRs

**Pull request closed without merging**